arctic/store/metadata_store.py

```diff
@@ -0,0 +1,229 @@
+        -------
+        Document written
+        """
+        document = {'_id': bson.ObjectId()}
+        document['symbol'] = symbol
+        document['metadata'] = metadata
+        document['start_time'] = start_time
+
+        self.insert_one(document)
+
+        logger.debug('Finished writing metadata for %s', symbol)
+
+        return document
+
+    def write_metadata_history(self, collection):
+        """
+        Manually overwrite entire metadata history for symbols in `collection`
+
+        Parameters
+        ----------
+        collection : `dict`
+            {symbol: (list of metadata, list of timestamps)}
+            symbol : `str`
+                symbol name for the item
+            metadata : `dict`
+                to be persisted
+            timestamp : `datetime.datetime`
+                start_time of the corresponding metadata
+            Example:
+                {'example': ([{}], [datetime.utcnow()])
+        """
+        if not isinstance(collection, dict):
+            raise TypeError('collection must be a dictionary.')
+        documents = []
+        for symbol, (entries, times) in collection.items():
+            if len(entries) != len(times):
+                raise ValueError('Number of entries and number of time stamps do not match.')
+            if self.has_symbol(symbol):
+                self.delete_all_metadata(symbol)
+            doc = {'symbol': symbol, 'metadata': entries[0], 'start_time': times[0]}
+            for metadata, start_time in zip(entries[1:], times[1:]):
+                if metadata == doc['metadata']:
+                    continue
+                doc['end_time'] = start_time
+                documents.append(doc)
+                doc = {'symbol': symbol, 'metadata': metadata, 'start_time': start_time}
+            else:
+                documents.append(doc)
+
+        self.insert_many(documents)
+
+    def write_metadata(self, symbol, metadata, start_time=None):
+        """
+        Update metadata entry for `symbol`
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name for the item
+        metadata : `dict`
+            to be persisted
+        start_time : `datetime.datetime`
+            when metadata becomes effective
+            Default: datetime.utcnow()
+        """
+        if start_time is None:
+            start_time = dt.utcnow()
+        old_metadata = self.find_one({'symbol': symbol}, sort=[('start_time', pymongo.DESCENDING)])
+        if old_metadata is not None:
+            if old_metadata['start_time'] >= start_time:
+                raise ValueError('start_time={} is earlier than the last metadata @{}'.format(start_time,
+                                                                                              old_metadata['start_time']))
+            if old_metadata['metadata'] == metadata:
+                logger.warning('No change to metadata')
+                return metadata
+        elif metadata is None:
```

Comment:
Pass `metadata=None` will actually set `metadata` to be `None` which is intended as some user may find it useful to unset metadata for a period of time. This will also explain the next two comments.